js/src/ui/TxList/TxRow/txRow.js

```diff
@@ -165,6 +262,61 @@ class TxRow extends Component {
 
     return `/addresses/${address}`;
   }
+
+  getCondition = () => {
+    const { blockNumber, tx } = this.props;
+    let { time, block } = tx.condition;
+
+    if (time) {
+      if ((time.getTime() - Date.now()) >= 0) {
+        return `${dateDifference(new Date(), time, { compact: true })} left`;
+      } else {
+        return 'submitting...';
+      }
+    } else if (blockNumber) {
+      block = blockNumber.minus(block);
+      return (block.toNumber() < 0)
+        ? block.abs().toFormat(0) + ' blocks left'
+        : 'submitting...';
+    }
+  }
+
+  cancelTransaction = () => {
+    const { parity } = this.context.api;
+    const { hash } = this.props.tx;
+
+    parity.removeTransaction(hash);
+    this.setState({ canceled: true });
+  }
+
+  editTransaction = () => {
+    const { parity } = this.context.api;
+    const { hash, gas, gasPrice, to, from, value, input, condition } = this.props.tx;
+
+    parity.removeTransaction(hash);
```

Comment:
I would consider the fact that you can't use the same gas price as a bug. In this case, you can still use the same gas price, which is automated anyways. If the user didn't have the choice for gas price, but rather the GP was automated, than I would just GP++ and purely edit.